### PR TITLE
Sort

### PR DESCRIPTION
**Pull request by nonoakij**
## Summary

Fixed a bug in sort.

https://github.com/cam-inc/viron/assets/45055030/f017b292-c938-416a-a146-1b66e8ffc3de

## Checklist
- [x] [changeset file(s)](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) included

### CODE REVIEW

File: packages/app/src/components/table/index.tsx [hunk 1]
```diff
@@ -64,17 +41,37 @@ const Table: React.FC<Props> = ({
...
+                              case SORT.DESC:
+                                sort = SORT.NONE;
+                                break;
+                              case SORT.NONE:
+                                sort = SORT.ASC;
+                                break;
+                            }
+                            onRequestSortChange?.(column.key, sort);
+                          }}
+                        >
```
Comment (nonoakij) [inline packages/app/src/components/table/index.tsx:65]: Warp with button only when sorting is possible

File: packages/app/src/pages/endpoints/[endpointId]/_/content/types/table/index.tsx [hunk 2]
```diff
@@ -31,14 +35,9 @@ const ContentTable: React.FC<Props> = ({
...
   onDescendantOperationSuccess,
   onDescendantOperationFail,
+  sortState,
 }) => {
-  const [sorts, setSorts] = useState<
-    Record<
-      TableProps['columns'][number]['key'],
-      TableProps['columns'][number]['sort']
-    >
-  >({});
```
Comment (nonoakij) [inline packages/app/src/pages/endpoints/[endpointId]/_/content/types/table/index.tsx:40]: If sort is changed, this component itself is unmonted and cannot retain State.

File: packages/app/src/utils/oas/index.ts [hunk 3]
```diff
@@ -786,6 +786,10 @@ export const mergeTableSortRequestValue = (
         break;
     }
   }
+  // HOTFIX: 空の配列をふくまないようにする
+  if (!requestValue.parameters?.sort.length) {
+    delete requestValue.parameters?.sort;
+  }
```
Comment (nonoakij) [inline packages/app/src/utils/oas/index.ts:792]: When sort is none, query is as follows:
`?sort=`
In this case, the API may not support it, so it must be removed.

File: packages/app/src/components/table/index.tsx [hunk 4]
```diff
@@ -179,37 +174,44 @@ const ThTitle: React.FC<ThTitleProp> = ({ on, column, onClick }) => {
...
+          </div>
+        )}
       </div>
-      <Popover {...popover.bind}>
-        <div className="w-max text-thm-on-surface">
-          {column.schema.description}
-        </div>
-      </Popover>
+      {/*
+      BUG: If it is smaller than lg, the popover will repeatedly show and hide.
```
Comment (h8570rg) [inline packages/app/src/components/table/index.tsx:205]: 🤔

File: packages/app/src/pages/endpoints/[endpointId]/_/content/parts/body/index.tsx [hunk 5]
```diff
@@ -37,6 +38,16 @@ const Body: React.FC<Props> = ({
     console.log(error);
   }, []);
 
+  // HOTFIX: TableContent 内部でStateを持つと、sort の値を変更した際に
+  // TableContent が unmount されて state を保持できないため、
+  // ここで State を持つ
```
Comment (h8570rg) [inline packages/app/src/pages/endpoints/[endpointId]/_/content/parts/body/index.tsx:43]: Japanese

**Review by h8570rg — APPROVED** [review]

**Pull request merged**